docs/proposals/image-signing.md

```diff
@@ -0,0 +1,129 @@
+# Image Signing
+
+## Abstract
+
+This proposal describes the process of verifying the signatures of Docker Images and
+enforce enforce policy that will prevent the image from running in the cluster if the
+image signature is not trusted.
+
+## Use-cases
+
+* As an admin, I want to be able to add trusted GPG public keys to my cluster and determine through policy
+  the keys that are required to run certain images
+* As an user, I want to be able to see what images are signed and if the signature is trusted
+
+## Overview
+
+### Signing Images with Skopeo
+
+The current signing mechanism for Docker image is to use the
+[Skopeo](https://github.com/projectatomic/skopeo/blob/master/docs/skopeo.1.md) tool which
+will sign the Docker image locally, using the GPG key and attach the signature to the
+Docker Image. The Docker image is then pushed to a Docker Registry with enabled signature
+storage.
+
+Signature verification [policy
+files](https://github.com/containers/image/blob/master/docs/policy.json.md) are used to
+define policy, e.g. trusted keys, applicable when deciding whether to accept an image, or
+individual signatures of that image, as valid:
+
+```json
+{
+  "default": [{"type": "reject"}],
+  "transports": {
+    "atomic": {
+      "172.30.62.231:5000": [
+      {
+        "type": "signedBy",
+        "keyType": "GPGKeys",
+        "keyPath": "/keys/public.gpg"
+      }
+      ]
+    },
+    "docker": {
+      "docker.io/openshift": [{"type": "insecureAcceptAnything"}]
+    }
+  }
+}
+```
+
+In this example, the default policy is set to reject all Docker images by default and
+whitelist only images that are pushed in the OpenShift Integrated Registry and signed
+using the public GPG key.
+It also whitelists all Docker images from the OpenShift DockerHub repository.
+
+To sign an Docker image, you have to execute the following command:
+
+```console
+$ skopeo --tls-verify=false copy --sign-by user@example.com docker://docker.io/myrepo/image:latest \
+  atomic:172.30.74.246:5000/mynamespce/image:latest
+```
+
+When this command is executed, skopeo will ask you to enter the password the
+*user@example.com* GPG key, sign the local image and push the signed image into OpenShift
+Integrated Registry.
+
+### Kubernetes `ImagePolicyWebhook` Admission Controller
+
+Kubernetes provides the
+[ImagePolicyWebhook](http://kubernetes.io/docs/admin/admission-controllers/#imagepolicywebhook)
+to allow a backend webhook to make admission decisions about whether an image specified by
+the container inside a Pod is allowed to run in the cluster or not.
+
+To configure this you have to specify a backed HTTPS service that Kubernetes will send the
+`ImageReview` objects as a payload, expecting the `ImageReview` in return with the
+*status* field set to indicate allowance or rejection of the image(s).
+
+## Assumptions and Constraints
+
+The assumption is that we provide a HTTPS backend that implements the `ImagePolicyWebhook`
```

Comment:
Include pros and cons as well.  

1.  alpha state of upstream admission is subject to change
1.  we want a consistent policy story and have existing configurable policy
1.  http server is an extra component which complicates administration